extension/goext/error.go

```diff
@@ -15,43 +15,192 @@
 
 package goext
 
-import "fmt"
-
-// ErrorCode identifies an error code
-type ErrorCode int
+import (
+	"fmt"
+	"net/http"
+	"runtime"
+)
 
-// Error represents an error
+// Error represents an error code with related HTTP status
 type Error struct {
-	Code ErrorCode
-	Err  error
+	Err    error
+	Status int
+	Origin string
 }
 
 // NewError returns a new error
-func NewError(code ErrorCode, err error) Error {
-	return Error{
-		Code: code,
-		Err:  err,
+func NewError(status int, err error) *Error {
+	return &Error{
+		Err:    err,
+		Status: status,
+		Origin: captureOrigin(),
+	}
+}
+
+func captureOrigin() string {
+	const SkipStackFrames = 3
+	// 0: captureOrigin()
+	// 1: NewError(...)
+	// 2: NewErrorXXX(...)
+	// 3: some_extension.go:123
+	_, file, line, ok := runtime.Caller(SkipStackFrames)
+	if !ok {
+		return "<unknown>"
+	}
+	return fmt.Sprintf("%s:%d", file, line)
+}
+
+type ErrorMatcher struct {
+	Expected *Error
+}
+
+func (matcher *ErrorMatcher) Match(actual interface{}) (bool, error) {
```

Comment:
how is the Match function supposed to be used? please add some UTs

Reply:
@p-kozlowski utest added, docs added

Reply:
Adding Equal function will cause import of gomega in goext package. I don't think it's a good idea. What do you think about this issue?

Reply:
I agree that we shouldn't mix test and production code. Perhaps we should move all test-related stuff (like the `ErrorMatcher` type) to a separate package. If someone uses `ErrorMatcher`, it's more than likely they're using gomega as well, so it should not be an issue to import it from the test package. Or would it?

Reply:
Good point, added exported test package for goext